### ISSUE TITLE
Start publishing the rust bindings to crates.io

### ISSUE THREAD

**Issue by ckipp01** [tree-sitter/tree-sitter-scala#193]
### Discussed in https://github.com/tree-sitter/tree-sitter-scala/discussions/191

<div type='discussions-op-text'>

<sup>Originally posted by **smarter** February 18, 2023</sup>
Once https://github.com/tree-sitter/tree-sitter-scala/pull/190 is merged, we'll be able to compile the Rust bindings. It'd be great to then publish them so they can be used in downstream project such as https://github.com/github/stack-graphs/ (which github is apparently starting to use for go-to-definition/find-all-references: https://docs.github.com/en/repositories/working-with-files/using-files/navigating-code-on-github#precise-and-search-based-navigation)</div>

**Comment by ckipp01** [comment]
Hey @smarter, thanks for bringing this up! I moved this over to an issue right away since we can totally do this. I'm not 100% sure on the process and ownership since I see that this hasn't been published at all yet. I'll have to take a look at the publishing process to see how we can get the set up and hopefully automated.

**Comment by eed3si9n** [comment]
It might be worth considering the version scheme that can express tree-sitter version number vs our own tree-sitter-scala version number.

For example, we could do something like:
- 0th version given tree-sitter 0.20.x (currently [0.20.9](https://crates.io/crates/tree-sitter)): **0.20.0**
- 1th version given tree-sitter 0.20.x: **0.20.1**

This way the user can write:

```
[dependencies]
tree-sitter = "0.20"
tree-sitter-scala = "0.20"
```

or lock it down to:

```
[dependencies]
tree-sitter = "0.20.9"
tree-sitter-scala = "0.20.0"
```

**Comment by eed3si9n** [comment]
I back published 0.20.0 - https://crates.io/crates/tree-sitter-scala/
